src/utils/calculator.ts

```diff
@@ -0,0 +1,39 @@
+import { GatewayOracle } from 'ardrive-core-js';
+import type { ArweaveOracle } from 'ardrive-core-js/lib/arweave_oracle';
+import regression, { DataPoint } from 'regression';
+
+export class Calculator {
+	private static readonly sampleByteVolumes = [
+		100000, // 100KB
+		100000000, // 100MB
+		10000000000 // 10GB
+	];
+	private sampleWinstonValues: number[] = [];
+	private regressionInstance?: regression.Result;
+
+	constructor(private oracle: ArweaveOracle = new GatewayOracle()) {}
+
+	public async setup(): Promise<void> {
```

Comment:
In the interest of making this class "easy to use correctly and hard to use incorrectly", I'd suggest the following:
1) Take an optional parameter in the constructor, e.g. `deferSetup` which, if true, SKIPS setup upon construction. This would allow you to create an instance without kicking off network requests. Otherwise, however, setup() would be called upon construction.
2) Make setup a private async call that returns a Promise. Keep private state for whether or not setup has taken place and whether or not setup is in progress. When getPriceForBytes is called, await on the call to setup, then run code. Internally, setup should check whether setup has completed previously. If so, the promise should resolve immediately. If not, then wait for the previous setup context to finish running (so we don't make multiple network requests) and call setTimeout to wait for, say, 50ms after which time we will check whether setup has completed in the previous setup context, if not, repeat (later on we can add a max wait time so we don't blow up the stack). If it has completed, resolve. If setup has never run, then await the results of the call to fetchData().

Reply:
I did it a little different: instead of using the timer I'm directly waiting for the promise to finish. Check it out

Reply:
I called it "setupWorker"